### DISCUSSION TITLE
OTP Issue on Android using this package.

### DISCUSSION THREAD

**Discussion by symmetrically** [invertase/react-native-firebase#4395] [Q&A] [answered]
We have React Native app where we are using Firebase to authenticate user and verify using OTP.  We are using this package for the implementation. We are facing two issues which might be different from each other but could possibly be same. Here is what is happening:

1. A valid phone number where the sim card of that phone number is inserted works fine with OTP on Android. No issues here
2. A valid phone number where the sim card is inserted on a different phone gets the OTP but when you enter it manually it does not do anything. The screen stays stuck.
3. A valid phone number which is ported to different carrier does not get OTP at all regardless the sim is inserted in the same phone or not.

Happy to provide more information to solve this problem. Any help will be highly appreciated

**Comment by mikehardy** [comment, marked as answer]
I think this will be a problem between firebase and the carriers, where the carrier does not send these lower-priority messages with regularity. There are 3 main carriers in my country for instance, 1 passes these OTPs with delay, 1 doesn't pass them for ported numbers, and 1 never passes them. My resolution? OTPs are not a reliable form of anything. They are interesting and may *augment* your solution, but must not be relied on as *the* solution for anything.

**Reply by mikehardy** [reply]
And it's nothing related to react-native-firebase unfortunately - there is nothing this module can do.
